server/game/effectengine.js

```diff
@@ -87,7 +87,7 @@ class EffectEngine {
     }
 
     onCardMoved(event) {
-        this.unapplyAndRemove(effect => effect.duration === 'persistent' && effect.source === event.card && (effect.location === event.originalLocation || event.parentChanged && effect.location !== 'any'));
+        this.unapplyAndRemove(effect => effect.duration === 'persistent' && effect.source === event.card && event.originalLocation !== event.newLocation && (effect.location === event.originalLocation || event.parentChanged && effect.location !== 'any'));
```

Comment:
good catch. what i am trying to do is the following:
a card with a persistent effect like green-apple knight is facedown attached to a fool via A Mummers Farce in addition to another knight in play. the effect (STR increase) is not active while the character is facedown. as soon as the knight is marshalled, the onCardMoved event is fired, but instead of activating the effect, the effect is removed and no STR increase happens.
I will build a test for it... i think that will figuring this out easier in the future.

i changed the condition in effectEngine now to check for `!event.facedownChanged` and pass facedownChanged to the event in basecard.